arch/posix/CMakeLists.txt

```diff
@@ -1,5 +1,12 @@
 # SPDX-License-Identifier: Apache-2.0
 
+execute_process(COMMAND arch OUTPUT_VARIABLE BUILD_ARCH OUTPUT_STRIP_TRAILING_WHITESPACE)
```

Comment:
Would you be so kind as to add a comment around here describing what this is doing?
I'm hoping for something like:
`# Detect the host architecture, so we can set the appropriate compile options flags depending also on the selected native_posix word size.
Otherwise this change adds to the mystic feel of our cmake files